PrimeNumbers/ProcessorInfo.h

```diff
@@ -92,37 +92,56 @@ void GetProcessorInfo(int* processorCount, int* processorGroupCount)
 /// <param name="processorCount"></param>
 /// <param name="isMultiCpuGroup"></param>
 /// <param name="threadHandles"></param>
-void SetThreadAffinity(int processorCount, bool isMultiCpuGroup, std::vector<HANDLE>& threadHandles)
+void SetThreadAffinity(int processorCount, 
+					   int numGroups, 
+	                   std::vector<HANDLE>& threadHandles, 
+	                   bool affinitizeEveryOtherProc)
 {
 	assert(threadHandles.size() == processorCount);
 
+	bool isMultiCpuGroup = (numGroups > 1);
+
 	for (int procNo = 0; procNo < processorCount; procNo++)
 	{
-		GroupProcNo groupProcNo(procNo);
+		int adjustedProcNo = (affinitizeEveryOtherProc ? (procNo * 2) : procNo);
+
+		//GroupProcNo groupProcNo(numGroups, adjustedProcNo);
 
 		if (isMultiCpuGroup)
 		{
 			GROUP_AFFINITY ga;
-			ga.Group = (WORD)groupProcNo.GetGroup();
+			//ga.Group = (WORD)groupProcNo.GetGroup();
+			ga.Group = (WORD)(adjustedProcNo >> 6);
 			ga.Reserved[0] = 0; // reserve must be filled with zero
 			ga.Reserved[1] = 0; // otherwise call may fail
 			ga.Reserved[2] = 0;
-			ga.Mask = (size_t)1 << groupProcNo.GetProcIndex();
+			//ga.Mask = (size_t)1 << groupProcNo.GetProcIndex();
+			ga.Mask = (size_t)1 << (adjustedProcNo % 64);
```

Comment:
what is `% 64` for? Is it just 64-bit or is it hardcoded for 64 core or something?

Reply:
the original code didn't work so I just did a quick hack here to run on the specific machine. it wouldn't run on a machine with cpu groups that has < 64 procs/group. feel free to fix this a proper way of course